pkg/job/job.go

```diff
@@ -36,6 +38,23 @@ func ConstructDockerJob(
 	jobInputs := []storage.StorageSpec{}
 	jobOutputs := []storage.StorageSpec{}
 
+	for _, inputURL := range inputUrls {
+		// split using LastIndex to support port numbers in URL
+		lastInd := strings.LastIndex(inputURL, ":")
+		rawURL := inputURL[:lastInd]
+		path := inputURL[lastInd+1:]
+		// should loop through all available storage providers?
```

Comment:
So there might be a future when we have S3 URLs and use a different (somehow authenticated) storage driver that can download them - I suppose at that point we would switch on the engine used - but at present we assume ipfs for inputVolumes and urldownload for inputUrls and this is OK :-)